### PR TITLE
Settings tests

### PR DESCRIPTION
**Pull request by hamsun7**

### CODE REVIEW

File: src/app/_helpers/kingdom.interceptor.ts [hunk 1]
```diff
@@ -1,67 +1,75 @@
...
-                x: 12,
-                y: 24,
-              },
-            }, status: 200,
-          }));
-        observer.complete();
-      });
-    }
+  Observable<HttpSentEvent | HttpHeaderResponse | HttpProgressEvent | HttpResponse<any> | HttpUserEvent<any>> {
+    console.log(req);
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: please remove the log

File: src/app/_helpers/kingdom.interceptor.ts [hunk 2]
```diff
@@ -1,67 +1,75 @@
...
+            observer.next(new HttpResponse<Kingdom>(
+              {
+                  body: kingdom,
+                  status: 200,
+              }));
+            observer.complete();
+          });
+        }
+        else if (req.url.endsWith('/kingdom') && (req.method === 'PUT')) {
+          console.log(req);
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: here too

File: src/app/_helpers/kingdom.interceptor.ts [hunk 3]
```diff
@@ -1,67 +1,75 @@
...
+          kingdom.name = req.body;
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: this doesn't do anything

File: src/app/app.module.ts [hunk 4]
```diff
@@ -52,10 +53,12 @@ import { KingdomSettingsComponent } from './kingdom-settings/kingdom-settings.co
...
 
   providers: [
-    { provide: HTTP_INTERCEPTORS, useClass: TokenInterceptor, multi: true },
-    { provide: HTTP_INTERCEPTORS, useClass: ResourceInterceptor, multi: true },
-    { provide: HTTP_INTERCEPTORS, useClass: LoginInterceptor, multi: true },
-    { provide: HTTP_INTERCEPTORS, useClass: BuildingsInterceptor, multi: true },
+//    { provide: HTTP_INTERCEPTORS, useClass: TokenInterceptor, multi: true },
+//    { provide: HTTP_INTERCEPTORS, useClass: ResourceInterceptor, multi: true },
+ //   { provide: HTTP_INTERCEPTORS, useClass: LoginInterceptor, multi: true },
+  //  { provide: HTTP_INTERCEPTORS, useClass: BuildingsInterceptor, multi: true },
```
Comment (balazsbarni) [inline src/app/app.module.ts]: uncomment these please

File: src/app/kingdom-settings/kingdom-settings.component.html [hunk 5]
```diff
@@ -1,5 +1,6 @@
-<form>
+<form (ngSubmit)="updateSettings(name)">
 <div>
+  <p>{{name}}</p>
```
Comment (balazsbarni) [inline src/app/kingdom-settings/kingdom-settings.component.html]: this is also not needed

File: src/app/kingdom-settings/kingdom-settings.component.ts [hunk 6]
```diff
@@ -17,8 +17,15 @@ export class KingdomSettingsComponent implements OnInit {
   }
 
   showSettings(): void {
-    this.name = this.settingsService.getSettings()
-      .subscribe(response => this.name = response.body.name);
+    this.settingsService.getSettings().subscribe(response => (
+      console.log(response),
```
Comment (balazsbarni) [inline src/app/kingdom-settings/kingdom-settings.component.ts]: log

File: src/app/kingdom-settings/kingdom-settings.component.ts [hunk 7]
```diff
@@ -17,8 +17,15 @@ export class KingdomSettingsComponent implements OnInit {
...
+      this.name = response.body.name));
+  }
+  updateSettings(name): void {
+    this.settingsService.updateSettings(name).subscribe(response => {
+      console.log(response);
```
Comment (balazsbarni) [inline src/app/kingdom-settings/kingdom-settings.component.ts]: logs

File: src/app/_helpers/kingdom.interceptor.ts [hunk 8]
```diff
@@ -1,67 +1,75 @@
...
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpResponse,
+  HttpSentEvent, HttpHeaderResponse, HttpProgressEvent, HttpUserEvent } from '@angular/common/http';
 import { Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
 import { Kingdom } from '../_models/kingdom.model';
+
+const kingdom: Kingdom = {
+  'id': 213,
+  name: 'My Kingdom',
+  'user_id': 22,
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: `userId`

File: src/app/_helpers/kingdom.interceptor.ts [hunk 9]
```diff
@@ -1,67 +1,75 @@
...
+  'buildings': [
+    {
+      'id': 123,
+      'type': 'townhall',
+      'level': 1,
+      'hp': 100,
+      'started_at': 1231232312,
+      'finished_at': 7652146122
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: `startedAt` `finishedAt`, just like in the `Hero` model

File: src/app/_helpers/kingdom.interceptor.ts [hunk 10]
```diff
@@ -1,67 +1,75 @@
...
+  ],
+  'troops': [
+    {
+      'id': 123,
+      'level': 1,
+      'hp': 100,
+      'attack': 50,
+      'defence': 20,
+      'started_at': 1231232312,
+      'finished_at': 7652146122
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: `startedAt` `finishedAt`, just like in the `Hero` model

File: src/app/_helpers/kingdom.interceptor.ts [hunk 11]
```diff
@@ -1,67 +1,75 @@
-import { HttpInterceptor,
-  HttpRequest,
-  HttpHandler,
-  HttpResponse,
-  HttpSentEvent,
-  HttpHeaderResponse,
-  HttpProgressEvent,
-  HttpUserEvent } from '@angular/common/http';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpResponse,
```
Comment (balazsbarni) [inline src/app/_helpers/kingdom.interceptor.ts]: this file shouldn't be in the `interceptors` folder?

File: src/app/kingdom-settings/kingdom-settings.service.ts [hunk 12]
```diff
@@ -13,6 +13,10 @@ export class KingdomSettingsService {
   constructor(private kingdomService: KingdomService) { }
 
   getSettings(): Observable<any> {
+    console.log(this.kingdomService.getKingdom());
```
Comment (balazsbarni) [inline src/app/kingdom-settings/kingdom-settings.service.ts]: log

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**